src/coreclr/src/jit/codegenlinear.cpp

```diff
@@ -1194,7 +1194,7 @@ void CodeGen::genUnspillRegIfNeeded(GenTree* tree)
             unsigned regCount = unspillTree->GetMultiRegCount();
             for (unsigned i = 0; i < regCount; ++i)
             {
-                genUnspillRegIfNeeded(unspillTree, i);
+                genUnspillRegIfNeeded(tree, i);
```

Comment:
Could you please clarify a bit why we need to use `RELOAD` tree here? It is not clear from the surrounding code.

Reply:
The reason is that when we unspill (reload) a tree, we need to reload it to the expected register. By default, we reload to the same register that it was spilled from, but in the case that the register doesn't meet the requirements of the consuming node, or is unavailable, a `RELOAD` tree will be inserted, and that will have the actual register we should reload into.

So you think this needs a comment? I thought it would be clear from context, but perhaps not.

Reply:
Thank you for the explanation. Do not we have the same issue in the previous condition 
`else if (unspillTree->IsMultiRegLclVar())`: `regNumber reg         = lclNode->GetRegNumByIdx(i);` , should not it be something like:
```
if (tree->gtOper == GT_RELOAD)
   reg = tree->GetRegNumByIdx(I); // get target reg from reload tree;
else
   reg = lclNode->GetRegNumByIdx(I); // no reload node, unspill to the original location.
```

> So you think this needs a comment? I thought it would be clear from context, but perhaps not.

Up to you, it is a bit unexpected to see `tree` use surrounded by `unspillTree` checks, but now we can do git annotate, find this comment and check that is not a typo.

Reply:
The case above is different because we never have RELOAD trees above reg-candidate lclVars. However, I can see that this is all a but convoluted so I've added a couple of asserts for the reg-candidate lclVar cases, and additional comments.